src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -37,8 +37,21 @@
       DetermineProjectJsonPath;
     </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
-
-  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
+  
+  <!--This target is required to evaluate this property uniquely in the context of each project -->
+  <Target Name="EvaluateGenerateTestProjectJsonDependsOn" >
+    <PropertyGroup>
+      <GenerateTestProjectJsonDependsOn Condition="'$(SkipGenerateTestProjectJson)'=='false'">
```

Comment:
This shouldn't be in a wrapper target. Also I would change this condition to `'$(SkipGenerateTestProjectJson)' == 'true'`